helpers/crearMovimientoDetalle.go

```diff
@@ -0,0 +1,311 @@
+						logs.Error(err)
+						panic(errorctrl.Error("CrearMovimientoDetalle - json.Marshal(detalleNuevoMov)", err2, "500"))
+					}
+
+					nuevoMovimiento := models.MovimientoProcesoExterno{
+						TipoMovimientoId:         result.TipoMovimientoId,
+						ProcesoExterno:           result.ProcesoExterno,
+						MovimientoProcesoExterno: result.MovimientoProcesoExterno,
+						Activo:                   result.Activo,
+						Detalle:                  string(detalleNuevoMovStr),
+					}
+
+					// logs.Debug("NUEVO MOVIMIENTO: ", &nuevoMovimiento)
+					if _, err := models.AddMovimientoProcesoExterno(&nuevoMovimiento); err != nil {
+						panic(errorctrl.Error("CrearMovimientoDetalle - models.AddMovimientoProcesoExterno(result)", err, "500"))
+					}
+				} else if estado["Estado"].(string) != "Preliminar" {
+					err := "No se reconoce el estado del movimiento proceso externo"
+					panic(errorctrl.Error("crearMovimientoDetalle - estado[\"Estado\"].(string) != \"Preliminar\"", err, "500"))
+				}
+			}
+		}
+	}
+
+	saldo := cuentaMovimientoDetalle.Saldo
+	valor := cuentaMovimientoDetalle.Valor
+
+	if !publicar {
+		if saldo == 0 && valor == 0 {
+			err := "Tanto el saldo como el valor tienen un valor de 0, no se puede añadir el movimiento detalle"
+			panic(errorctrl.Error("crearMovimientoDetalle - saldo == 0 && valor == 0", err, "400"))
+		} else if saldo != 0 && valor != 0 {
+			err := "Tanto el saldo como el valor tienen un valor diferente de 0, no se puede añadir el movimiento detalle"
+			panic(errorctrl.Error("crearMovimientoDetalle - saldo != 0 && valor != 0", err, "400"))
+		}
+	} else {
+		if saldo != valor {
+			err := "El saldo y el valor de publicación debe ser el mismo"
+			panic(errorctrl.Error("crearMovimientoDetalle - saldo != valor", err, "400"))
+		}
+	}
+
+	detalleCuenPre := cuentaMovimientoDetalle.Cuen_Pre
+
+	if detalleCuenPre == "" {
+		err := "No se han ingresado datos de cuentas para crear movimientos detalle"
+		panic(errorctrl.Error("crearMovimientoDetalle - detalleCuenPre == \"\"", err, "400"))
+	}
+
+	// logs.Debug("INSERTAR movimiento: CrearMovimientoDetalle", idMovProcExterno)
+
+	if registroMovimientoDetalle, err := RegistroMovimientoDetalle(detalleCuenPre, idMovProcExterno, saldo, valor, publicar); err != nil {
+		logs.Error(err)
+		panic(err)
+	} else {
+		if result, err := models.AddMovimientoDetalle(&registroMovimientoDetalle); err != nil {
+			logs.Error(err)
+			panic(err)
+		} else {
+			resultCast := int(result)
+			if movimientoDetalleRegistrado, err = models.GetMovimientoDetalleById(resultCast); err != nil {
+				logs.Error(err)
+				panic(err)
+			}
+		}
+	}
+
+	return movimientoDetalleRegistrado, idNuevoMovProcExterno, nil
+}
+
+func RegistroMovimientoDetalle(detalleCuenPre string, idMovProcExterno string, saldo float64, valor float64, publicar bool) (registroMovimientoDetalleRespuesta models.MovimientoDetalle, outputError map[string]interface{}) {
```

Comment:
1. Linea muy larga
2. Falta documentar, cambiar los `...` en este _suggested change_
```suggestion
// RegistroMovimientoDetalle ...
func RegistroMovimientoDetalle(detalleCuenPre string, idMovProcExterno string,
saldo float64, valor float64, publicar bool) (
registroMovimientoDetalleRespuesta models.MovimientoDetalle,
outputError map[string]interface{}) {
```